### PR TITLE
samples: Bluetooth: df: Add DF peripheral sample

### PR DESCRIPTION
**Pull request by ppryga-nordic**
Add new sample application for direction finding to show
how the functionality works in connected mode for peripheral role.

Signed-off-by: Piotr Pryga <piotr.pryga@nordicsemi.no>

### CODE REVIEW

**Comment by ppryga** [comment]
The PR has dependency on #41905. It should be merged after the #41905.

**Review by alwa-nordic — APPROVED** [review]

File: samples/bluetooth/direction_finding_peripheral/src/main.c [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+#include <bluetooth/direction.h>
+
+#define DF_FEAT_ENABLED BIT64(BT_LE_FEAT_BIT_CONN_CTE_RESP)
+
+static const struct bt_data ad[] = {
+	BT_DATA_BYTES(BT_DATA_FLAGS, (BT_LE_AD_GENERAL | BT_LE_AD_NO_BREDR)),
+	BT_DATA_BYTES(BT_DATA_LE_SUPPORTED_FEATURES, BT_LE_SUPP_FEAT_24_ENCODE(DF_FEAT_ENABLED)),
+};
+
+/* Latency set to zero, to enforce PDU exchange every connection event */
```
Comment (kruithofa) [inline samples/bluetooth/direction_finding_peripheral/src/main.c:31]: Comment says zero, but define is 5?

File: samples/bluetooth/direction_finding_peripheral/src/main.c [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+#define CONN_LATENCY 5U
+/* Inteval used to run CTE request procedure periodically.
+ * Value is a number of connection events.
+ */
+#define CTE_REQ_INTERVAL (CONN_LATENCY + 10U)
+/* Length of CTE in unit of 8 us */
+#define CTE_LEN (0x14U)
+
+#if defined(CONFIG_BT_CTLR_DF_ANT_SWITCH_RX)
+static const uint8_t ant_patterns[] = { 0x2, 0x0, 0x5, 0x6, 0x1, 0x4, 0xC, 0x9, 0xE, 0xD, 0x8 };
```
Comment (kruithofa) [inline samples/bluetooth/direction_finding_peripheral/src/main.c:41]: is there an explanation somewhere of what the values mean?

File: samples/bluetooth/direction_finding_peripheral/src/main.c (hunk 1 above)
Comment (ppryga) [reply samples/bluetooth/direction_finding_peripheral/src/main.c:31]: Leftover, corrected to zero.

File: samples/bluetooth/direction_finding_peripheral/src/main.c (hunk 2 above)
Comment (ppryga) [reply samples/bluetooth/direction_finding_peripheral/src/main.c:41]: The numbers are arbitrary. They have a meaning if there is a particular antenna matrix used with sample. So numbers are vendor specific. Here are just some numbers that are used by Controller to change state of GPIOs.

**Review by kruithofa — APPROVED** [review]

**Review by jhedberg — APPROVED** [review]
Looks fine, but it seems some CI checks are failing..

**Comment by ppryga** [comment]
Fixed build issues.

**Comment by ppryga** [comment]
@jhedberg @cfriedt @nashif The PR is ready for merge

**Pull request merged**